src/server/pfs/server/driver.go

```diff
@@ -2301,8 +2328,22 @@ func (d *driver) createBranch(txnCtx *txnenv.TransactionContext, branch *pfs.Bra
 		// Determine if this is a provenance update
 		sameTarget := branch.Repo.Name == commit.Repo.Name && branch.Name == commit.ID
 		if !sameTarget && provenance != nil {
-			return fmt.Errorf("cannot point branch \"%s\" at target commit \"%s/%s\" without clearing its provenance",
-				branch.Name, commit.Repo.Name, commit.ID)
+			ci, err := d.inspectCommit(txnCtx.Client, commit, pfs.CommitState_STARTED)
+			if err != nil {
+				return err
+			}
+			for _, provC := range ci.Provenance {
+				if !branchContains(provenance, provC.Branch) {
```

Comment:
Could this interact badly with deferred downstream processing?

Reply:
That definitely sounds plausible, I'm just going to write a test that does extract restore on a deferred processing workload and see if I can get it to break.

Reply:
Did this, test seems to pass.